delta/html.py

```diff
@@ -258,6 +258,74 @@ def classes_check(op):
     return True
 
 
+def base_picture(root, op):
+    pic = op['insert'].get('picture')
+    attributes = pic['attributes']
+    raw_sources = pic['sources']
+
+    lazy_load = attributes.get('lazy-load')
+    srcset_key = 'data-srcset' if lazy_load else 'srcset'
+
+    figure_tag = sub_element(root, 'figure')
+
+    a = None
+    href = attributes.get('link')
+    if href:
+        a = sub_element(figure_tag, 'a')
+        a.attrib['href'] = href.get('url')
+        if href.get('openNewTab'):
+            a.attrib['target'] = '_blank'
+            a.attrib['rel'] = 'noopener noreferrer'
```

Comment:
Isn't a lot of this code reused from the `base_image` function? Is there a simple way to have a shared set of functions that they can both use?